exla/c_src/exla/mlir/builder.cc

```diff
@@ -30,6 +30,48 @@ mlir::Value MLIRFunction::AddOp(mlir::Value lhs, mlir::Value rhs) {
   return op;
 }
 
+mlir::Value MLIRFunction::MulOp(mlir::Value lhs, mlir::Value rhs) {
+  module_->builder()->setInsertionPointToEnd(&func_->getBody().back());
+  auto op = module_->builder()->create<mlir::mhlo::MulOp>(module_->builder()->getUnknownLoc(), lhs, rhs);
+  return op;
+}
+
+mlir::Value MLIRFunction::MinOp(mlir::Value lhs, mlir::Value rhs) {
+  module_->builder()->setInsertionPointToEnd(&func_->getBody().back());
+  auto op = module_->builder()->create<mlir::mhlo::MinOp>(module_->builder()->getUnknownLoc(), lhs, rhs);
+  return op;
+}
+
+mlir::Value MLIRFunction::MaxOp(mlir::Value lhs, mlir::Value rhs) {
+  module_->builder()->setInsertionPointToEnd(&func_->getBody().back());
+  auto op = module_->builder()->create<mlir::mhlo::MaxOp>(module_->builder()->getUnknownLoc(), lhs, rhs);
+  return op;
+}
+
+mlir::Value MLIRFunction::RemOp(mlir::Value lhs, mlir::Value rhs) {
+  module_->builder()->setInsertionPointToEnd(&func_->getBody().back());
+  auto op = module_->builder()->create<mlir::mhlo::RemOp>(module_->builder()->getUnknownLoc(), lhs, rhs);
+  return op;
+}
+
+mlir::Value MLIRFunction::PowOp(mlir::Value lhs, mlir::Value rhs) {
+  module_->builder()->setInsertionPointToEnd(&func_->getBody().back());
+  auto op = module_->builder()->create<mlir::mhlo::PowOp>(module_->builder()->getUnknownLoc(), lhs, rhs);
+  return op;
+}
+
+mlir::Value MLIRFunction::DivOp(mlir::Value lhs, mlir::Value rhs) {
+  module_->builder()->setInsertionPointToEnd(&func_->getBody().back());
+  auto op = module_->builder()->create<mlir::mhlo::DivOp>(module_->builder()->getUnknownLoc(), lhs, rhs);
+  return op;
+}
+
+mlir::Value MLIRFunction::Atan2Op(mlir::Value lhs, mlir::Value rhs) {
+  module_->builder()->setInsertionPointToEnd(&func_->getBody().back());
+  auto op = module_->builder()->create<mlir::mhlo::Atan2Op>(module_->builder()->getUnknownLoc(), lhs, rhs);
+  return op;
+}
```

Comment:
I wonder if it is worth macro-fying those.

Btw, why do we need to define those? Should we instead call these directly from exla.cc?

Reply:
I believe this is so we have access to the function `module_` without cluttering `exla.cc`.
Not sure about macros, though. Torchx has a (IMO) really high first-contribution barrier because of them.